packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -429,7 +429,7 @@ export class Table extends Construct {
       return;
     }
     principal.addToPolicy(new iam.PolicyStatement()
-      .addResource(this.tableArn)
+      .addResources(this.tableArn, new cdk.Token(() => this.hasIndex ? `${this.tableArn}/index/*` : new cdk.Aws().noValue).toString())
```

Comment:
Because this won't throw up IAM diffs to everyone who's already deployed an indexless table.

Reply:
ha! not a bad reason. thanks.

also, wondering perhaps `addResource` can just ignore `undefined`, so you don't need to specify `noValue`. A bit more "cdk-native", no?

Reply:
Agreed, although according to the typing `addResource()` takes a `string` and not a `string | undefined`. It would be annoying if a stringified Token would change its type halfway through processing...

(Nothing preventing you from doing that today, but I'm not sure our code would be robust against it).

Reply:
Got it. No worries.